### PR TITLE
JT-18 (add new stock feature)

### PR DESCRIPTION
**Pull request by fstk1337**

### CODE REVIEW

File: vue-app/src/api/stocks.js [hunk 1]
```diff
@@ -6,8 +6,14 @@ export default function (instance) {
         get(id) {
             return instance.get(`satellites/${id}`)
         },
-        update(payload) {
+        update(payload, id) {
+            return instance.put(`satellites/${id}`, payload)
```
Comment (pelmegov) [inline vue-app/src/api/stocks.js:10]: Minor: baseUrl to constant?

File: vue-app/src/components/AddStockForm.vue [hunk 2]
```diff
@@ -27,27 +34,69 @@ export default {
...
       }
       this.$load(async () => {
-        await this.$api.satellites.create(this.payload);
-        console.log('[SUCCESS]: Stock saved')
-      });
+        const result = await this.$api.stocks.create(this.payload);
+        if (result.status === 200) {
+          this.addResult = JSON.parse(result.request.response);
+          this.$emit('action', this.addResult);
+          this.messageClass = 'success';
```
Comment (pelmegov) [inline vue-app/src/components/AddStockForm.vue:47]: Minor: 3 actions incapsulate to one method and reuse it in different places.

File: vue-app/src/components/AddStockForm.vue [hunk 3]
```diff
@@ -27,27 +34,69 @@ export default {
...
+      this.messageText = error.response.data.message;
+      this.showMessage = true;
+    },
+    refresh: function() {
       this.name = '';
       this.description = '';
+      this.messageClass = '';
+      this.messageText = '';
+      this.showMessage = false;
+      document.getElementById('stock-description').removeAttribute("style");
```
Comment (pelmegov) [inline vue-app/src/components/AddStockForm.vue:65]: Why is it required?

**Review by pelmegov — APPROVED** [review]

File: vue-app/src/components/AddStockForm.vue (hunk 3 above)
Comment (fstk1337) [reply vue-app/src/components/AddStockForm.vue:65]: > Why is it required?

Refresh it's like an action to return component into default state (no message show, all fields are empty and textarea with a default height)

File: vue-app/src/api/stocks.js (hunk 1 above)
Comment (fstk1337) [reply vue-app/src/api/stocks.js]: > Minor: baseUrl to constant?

Okay, done.

File: vue-app/src/components/AddStockForm.vue (hunk 2 above)
Comment (fstk1337) [reply vue-app/src/components/AddStockForm.vue]: > Minor: 3 actions incapsulate to one method and reuse it in different places.

Done.

**Review by aalexeen — APPROVED** [review]

**Pull request merged**